src/image/p5.Image.js

```diff
@@ -657,7 +653,32 @@ p5.Image.prototype.mask = function(p5Image) {
   ];
 
   this.drawingContext.globalCompositeOperation = 'destination-in';
-  p5.Image.prototype.copy.apply(this, copyArgs);
+  if (this.gifProperties) {
+    const prevFrameData = this.drawingContext.getImageData(
+      0,
+      0,
+      this.width,
+      this.height
+    );
+    for (let i = 0; i < this.gifProperties.frames.length; i++) {
+      this.drawingContext.clearRect(0, 0, this.width, this.height);
+      this.drawingContext.putImageData(
+        this.gifProperties.frames[i].image,
+        0,
+        0
+      );
+      p5.Image.prototype.copy.apply(this, copyArgs);
+      this.gifProperties.frames[i].image = this.drawingContext.getImageData(
+        0,
+        0,
+        this.width,
+        this.height
+      );
+    }
+    this.drawingContext.putImageData(prevFrameData, 0, 0);
```

Comment:
I tried drawing the current frame after the masking is done, but this causes the new test on the current frame to fail. This is surprising to me because I expect that if I put the masked frame using `putImageData` and then fetch the pixels using `get`, I should get the masked image back. Also, calling `img.reset` in the test (to draw the first frame) causes the test to pass. Any suggestions as to what might be going on?

Reply:
Are you still getting a test failure? I just tried running locally but it seems to work OK!

Reply:
the failing unit tests were due to #5772 .

Reply:
Ok great! If that was the only issue, I can take another look through today to make sure I haven't forgotten about anything, but then this PR could be good to go!